File: Daily Challenge/Java/Name Matching Score 17.7.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s1 = sc.next();
        String s2 = sc.next();
        int[] v1 = new int[s1.length()];
        int[] v2 = new int[s2.length()];
        int scVal = 0;

        for (int i = 0; i < Math.min(s1.length(), s2.length()); i++) {
            if (s1.charAt(i) == s2.charAt(i)) {
                v1[i] = v2[i] = 1;
                scVal += 2;
            }
        }

        for (int i = 0; i < s1.length(); i++) {
            for (int j = 0; j < s2.length(); j++) {
                if (s1.charAt(i) == s2.charAt(j) && v1[i] == 0 && v2[j] == 0) {
                    v1[i] = v2[j] = 1;
                    scVal += 1;
                    break;
                }
            }
        }

        System.out.println(scVal);
    }
}

File: Daily Challenge/Python/Name Matching Score 17.7.2025.py
s1 = input()
s2 = input()
v1 = [0] * len(s1)
v2 = [0] * len(s2)
sc = 0
for i in range(min(len(s1), len(s2))):
    if s1[i] == s2[i]:
        v1[i] = v2[i] = 1
        sc += 2
for i in range(len(s1)):
    for j in range(len(s2)):
        if s1[i] == s2[j] and not v1[i] and not v2[j]:
            v1[i] = v2[j] = 1
            sc += 1
print(sc)